clippy_lints/src/methods/unnecessary_min.rs

```diff
@@ -0,0 +1,153 @@
+use std::cmp::Ordering;
+
+use super::UNNECESSARY_MIN;
+use clippy_utils::diagnostics::span_lint_and_sugg;
+
+use clippy_utils::consts::{constant, Constant};
+use clippy_utils::source::snippet;
+use clippy_utils::{clip, int_bits, unsext};
+use hir::Expr;
+
+use rustc_errors::Applicability;
+use rustc_hir as hir;
+use rustc_lint::LateContext;
+
+use rustc_middle::ty::{self, IntTy};
+use rustc_span::Span;
+
+pub fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, _: &'tcx Expr<'_>) {
+    if both_are_constant(cx, expr) {
+        return;
+    }
+    one_extrema(cx, expr);
+}
+fn lint(cx: &LateContext<'_>, expr: &Expr<'_>, sugg: Span, other: Span) {
+    let msg = format!(
+        "`{}` is never greater than `{}` and has therefore no effect",
+        snippet(cx, sugg, "Not yet implemented"),
+        snippet(cx, other, "Not yet implemented")
+    );
+    span_lint_and_sugg(
+        cx,
+        UNNECESSARY_MIN,
+        expr.span,
+        &msg,
+        "try",
+        snippet(cx, sugg, "Not yet implemented").to_string(),
+        Applicability::MachineApplicable,
+    );
+}
+
+fn try_to_eval<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) -> (Option<Constant<'tcx>>, Option<Constant<'tcx>>) {
+    let (left, right) = get_both_as_expr(expr);
+    (
+        (constant(cx, cx.typeck_results(), left)),
+        (constant(cx, cx.typeck_results(), right)),
+    )
+}
+fn get_both_as_expr<'tcx>(expr: &'tcx Expr<'_>) -> (&'tcx Expr<'tcx>, &'tcx Expr<'tcx>) {
+    match expr.kind {
+        hir::ExprKind::MethodCall(_, left1, right1, _) => {
+            let left = left1;
+            let right = &right1[0];
+            (left, right)
+        },
+        _ => unreachable!("this function gets only called on methods"),
+    }
+}
+#[derive(Debug)]
+enum Extrema {
+    Minimum,
+    Maximum,
+}
+fn detect_extrema<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) -> Option<Extrema> {
+    let ty = cx.typeck_results().expr_ty(expr);
+
+    let cv = constant(cx, cx.typeck_results(), expr)?;
+
+    match (ty.kind(), cv) {
```

Comment:
Can use https://doc.rust-lang.org/nightly/nightly-rustc/clippy_utils/consts/enum.Constant.html#method.int_value here

Reply:
Oh i didnt know  this method existed.
I will look into that.
Thank you!

Reply:
I'm not sure how to implement that change.
I need to use the `ty.kind()` because i have to know how many bits the integer has.
If i use your proposal, than i only know if it is signed or unsigned and i can't use the math in the arms because it gives me a `i128` if it  is signed. (`unsext` wants `u128`)
Do i miss or misunderstand something here?
Thank you for your feedback. I appreciate it.